snap/snapcraft.yaml

```diff
@@ -0,0 +1,134 @@
+name: edgex-device-usb-camera
+base: core20
+license: Apache-2.0
+adopt-info: metadata
+
+architectures:
+  - build-on: amd64
+  - build-on: arm64
+
+grade: stable
+confinement: strict
+
+slots:
+  edgex-secretstore-token:
+    interface: content
+    content: edgex-secretstore-token
+    source:
+      write: [$SNAP_DATA/device-usb-camera]
+
+plugs:
+  device-config:
+    interface: content 
+    target: $SNAP_DATA/config/device-usb-camera
+
+apps:
+  rtsp-simple-server:
+    command: bin/rtsp-simple-server $SNAP_DATA/config/rtsp-simple-server.yml
+    daemon: simple
+    install-mode: disable
+    plugs: [network-bind]
+
+  device-usb-camera:
+    command: bin/device-usb-camera $CONFIG_PRO_ARG $CONF_ARG $REGISTRY_ARG
+    command-chain:
+      - bin/startup-env-var.sh
+    environment:
+      CONFIG_PRO_ARG: "--cp=consul.http://localhost:8500"
+      CONF_ARG: "--confdir=$SNAP_DATA/config/device-usb-camera/res"
+      REGISTRY_ARG: "--registry"
+      DEVICE_PROFILESDIR: "$SNAP_DATA/config/device-usb-camera/res/profiles"
+      DEVICE_DEVICESDIR: "$SNAP_DATA/config/device-usb-camera/res/devices"
+      SECRETSTORE_TOKENFILE: $SNAP_DATA/device-usb-camera/secrets-token.json
+      # This is needed by ffprobe to open libpulsecommon shared object file
+      LD_LIBRARY_PATH: $SNAP/usr/lib/$SNAPCRAFT_ARCH_TRIPLET/pulseaudio
+    daemon: simple
+    install-mode: disable
+    plugs: [network, network-bind, camera]
+
+
+parts:      
+  hooks:
+    source: snap/local/hooks
+    plugin: make
+    build-snaps: [go/1.17/stable]
+    override-build: |
+      cd $SNAPCRAFT_PART_SRC
+      make build
+      install -DT ./cmd/configure/configure $SNAPCRAFT_PART_INSTALL/snap/hooks/configure
+      install -DT ./cmd/install/install $SNAPCRAFT_PART_INSTALL/snap/hooks/install
+
+  rtsp-simple-server:
+    source:
+      - on amd64:
+          https://github.com/aler9/rtsp-simple-server/releases/download/v0.18.3/rtsp-simple-server_v0.18.3_linux_amd64.tar.gz
+      - on arm64:
+          https://github.com/aler9/rtsp-simple-server/releases/download/v0.18.3/rtsp-simple-server_v0.18.3_linux_arm64v8.tar.gz
+    plugin: dump
+    organize:
+      rtsp-simple-server: bin/rtsp-simple-server
+      rtsp-simple-server.yml: config/rtsp-simple-server.yml
+
+    
+  device-usb-camera:
+    after: [metadata]
+    source: .
+    plugin: make
+    build-packages: [git, libzmq3-dev, pkg-config]
+    build-snaps: [go/1.17/stable]
+    stage-packages: [libzmq5, ffmpeg, libpulse0]
+    override-build: |
+      cd $SNAPCRAFT_PART_SRC
+
+      # The existing header file misses several of the definitions needed for
```

Comment:
The existing header file from where? I checked my snapcraft lxc build container and the only `videodev2.h` file I could find is the one you're copying here.

Reply:
Not sure; probably from core20 (package linux-headers-5.4.0-110).

The go4vl library mentions v5.10.x as the compilation dependency. See https://github.com/vladimirvivien/go4vl#compilation-requirements

Reply:
Should no longer be an issue when we upgrade to core22.

Reply:
Switched to core22 and removed the sideloaded header.